packages/flutter_tools/test/general.shard/web/web_asset_server_test.dart

```diff
@@ -93,39 +99,37 @@ void main() {
       'Content-Type': 'text/html',
       'content-length': '28',
     });
-  }, overrides: <Type, Generator>{
-    FileSystem: () => fileSystem,
-    Platform: () => platform,
-    ProcessManager: () => FakeProcessManager.any(),
   });
 
-  testUsingContext('release asset server serves content from flutter root', () async {
-    final ReleaseAssetServer assetServer = ReleaseAssetServer(Uri.base);
+  testWithoutContext('release asset server serves content from flutter root', () async {
+    final ReleaseAssetServer assetServer = ReleaseAssetServer(Uri.base,
+      fileSystem: fileSystem,
+      platform: platform,
+      flutterRoot: '/flutter',
+      webBuilDirectory: 'build/web',
+    );
     fileSystem.file('flutter/bar.dart')
       ..createSync(recursive: true)
       ..writeAsStringSync('void main() { }');
     final Response response = await assetServer
       .handle(Request('GET', Uri.parse('http://localhost:8080/flutter/bar.dart')));
 
     expect(response.statusCode, HttpStatus.ok);
-  }, overrides: <Type, Generator>{
-    FileSystem: () => fileSystem,
-    Platform: () => platform,
-    ProcessManager: () => FakeProcessManager.any(),
   });
 
-  testUsingContext('release asset server serves content from project directory', () async {
-    final ReleaseAssetServer assetServer = ReleaseAssetServer(Uri.base);
+  testWithoutContext('release asset server serves content from project directory', () async {
+    final ReleaseAssetServer assetServer = ReleaseAssetServer(Uri.base,
+      fileSystem: fileSystem,
+      platform: platform,
+      flutterRoot: '/flutter',
+      webBuilDirectory: 'build/web',
+    );
     fileSystem.file('bar.dart')
       ..createSync(recursive: true)
       ..writeAsStringSync('void main() { }');
     final Response response = await assetServer
       .handle(Request('GET', Uri.parse('http://localhost:8080/bar.dart')));
 
     expect(response.statusCode, HttpStatus.ok);
-  }, overrides: <Type, Generator>{
-    FileSystem: () => fileSystem,
-    Platform: () => platform,
-    ProcessManager: () => FakeProcessManager.any(),
```

Comment:
was the process manager never getting called?

Reply:
I guess not, since these are now `testWithoutContext`. cool.

Reply:
Negatory, but we have a rule that a context injected FIleSystem requires a context Injected ProcessManager